### PR TITLE
build: add env var to skip bindgen

### LINKED ISSUE #6: add env var for optional protobuf compilation
Could you add an environment variable to make the protobuf compilation optional ?
I'm just thinking that for releases, there is no need to recompile the protobuf definitions.

Thanks for the library!

### PR DESCRIPTION
**Pull request by happysalada**
I've added the env var to skip bindgen.
let me know if anything

closes #6

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -10,6 +10,13 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     println!("cargo:rerun-if-changed=res/proto/google/cloud/texttospeech/v1/cloud_tts.proto");
     println!("cargo:rerun-if-changed=res/proto/google/cloud/texttospeech/v1beta1/cloud_tts.proto");
 
+    let skip_bindgen = env::var("GOOGLE_COGNITIVE_API_SKIP_BINDGEN")
```
Comment (adambezecny) [inline build.rs]: I would name this differently, maybe **SKIP_PROTO_FILES_COMPILATION**?

also while you want to skip to compile commands for proto compilation you want to preserve **Ok(())**

so rather then calling return you should have two IFs to skip respective code.

can you change accordingly? did you test it works properly (with and without specifying env variable)?

**Comment by happysalada** [comment]
Thanks for the quick feedback!
I've adjusted the env var name and tested with and without the environment variable.

File: build.rs [hunk 2]
```diff
@@ -10,6 +10,13 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     println!("cargo:rerun-if-changed=res/proto/google/cloud/texttospeech/v1/cloud_tts.proto");
     println!("cargo:rerun-if-changed=res/proto/google/cloud/texttospeech/v1beta1/cloud_tts.proto");
 
+    let skip_bindgen = std::env::var("SKIP_PROTO_FILES_COMPILATION")
```
Comment (adambezecny) [inline build.rs]: just to make it perfect can we also rename the variable **skip_proto_compilation** instead of s**kip_bindgen**. ?

thanks!

Comment (happysalada) [reply build.rs]: Ha, youre right! Just had another go at it

**Comment by happysalada** [comment]
@adambezecny could you give another round of review when you have a moment ?

**Comment by adambezecny** [comment]
hi,

technically the PR is now OK I guess. But it occurred to me what we actually want to achieve here. I realised build script actually contains instruction `cargo:rerun-if-changed` so GRPC stubs will not be regenerated evert time if the proto files have not changed. So what is the benefit of adding env variable to suppress it? In other words why exactly do you need this feature? please clarify.

regards,

Adam

**Comment by happysalada** [comment]
Thanks for coming back to me!
Upon building the project, the protobuf compilation occurs everytime. I'm not sure why. I'm fine with just using the generated protobuf files. That's why I want an env variable to just disable compilation upon building my binary that depends on this crate.

**Comment by adambezecny** [comment]
ok but I guess what you actually did is you cloned the library and you are compiling directly library project itself right? Will you actually notice this if you are using the library just as an ordinary dependency? In fact I used to put compiled artefacts into source code of library but most other libraries do not  take this approach (i.e. generated artiefacts are not included). In this regard including generated GRPC stubs is kind of bad practice (and I will not do it in the future should I ever write similar library :) ). I think what is happening at your side is that you have actually never version of tonic library and hence initially artefacts get  compiled differently. If you tried to compile for second time nothing would happen because `cargo:rerun-if-changed` will do its job. Is it so? Let me know if I misunderstood something.

**Comment by happysalada** [comment]
I am using the library as a normal dependency. I am however using a different build system (nix) that builds everything from source.
I am starting to understand that my use case is very particular. I can just use a fork of the project if you don't think that will be useful for other people. Thank you for the feedback!

**Comment by adambezecny** [comment]
No worries, I will merge this PR, give me some time to test, will come back to you.

**Comment by happysalada** [comment]
Thank you !

**Comment by adambezecny** [comment]
hi

PR has been merged and new version **0.2.1** released on crates.io. See also [here](https://github.com/jabber-tools/google-cognitive-apis/releases/tag/0.2.1). Enjoy :)

**Comment by happysalada** [comment]
Thank you!

**Pull request merged**